### DISCUSSION TITLE
Updated RetroScaler's N64 BlueRetro Adapter with the official BlueRetro firmware. Have I bricked it?

### DISCUSSION THREAD

**Discussion by Kopite11** [darthcloud/BlueRetro#827] [Q&A] [answered]
Hello! Made the mistake of updating this adapter with the official BlueRetro firmware without realizing that RetroScaler have it's own recompiled firmware.
It started to have some issues, so I also factory resetted it, now I can't flash any firmware to the device and it's acting all weird.
I can get it to connect with blueretro.io but the "Live Output" only shows the following:

```
Requesting Bluetooth Device...
Connecting to GATT Server...
Getting BlueRetro Service...
> Bluetooth Device disconnected
NetworkError
Argh! Couldn't connect to BlueRetro
```

and

```
Reading Api version...
Requesting Bluetooth Device...
Connecting to GATT Server...
Getting BlueRetro Service...
ABI version: 2
Init Cfg DOM...
Get Global Config CHRC...
ABI version: 2
Init Cfg DOM...
Get Global Config CHRC...
Reading Global Config...
Reading Global Config...
Argh! NotSupportedError: GATT operation failed for unknown reason.
Argh! NotSupportedError: GATT operation failed for unknown reason.
> Bluetooth Device disconnected
```

and (when trying to flash a firmware, the field to select the .bin never show up)

```
Requesting Bluetooth Device...
Connecting to GATT Server...
Getting BlueRetro Service...
app_ver_is_hw2: false app_name_is_hw2: false
Init Cfg DOM...
app_ver_is_hw2: false app_name_is_hw2: false
Init Cfg DOM...
> Bluetooth Device disconnected
```

Since this thing doesn't have a USB port, have I just bricked it?
Any help would be appreciated!

**Comment by darthcloud** [comment]
Let's forget about the web config for a moment.


Power on N64.
Hold the button on the adapter for 15 sec then release it.

Then give the adapter 1 min to get back.

After that minutes power off N64.

Power it on again.

Now try to connect a controller.

Does it work?

**Reply by Kopite11** [reply]
Yeah, tried that and now the adapter works, kinda...

The controller syncs, but almost everything I try to do on the web interface returns one of those errors mentioned.

Since I've done a factory reset on the device, now the web interface shows:
`Connected to: BlueRetro_N64_6BB2 (0c:8b:95:e6:6b:b2) [1]`

And as I mentioned, can't get it to update to the latest 1.9 firmware version...  the field to select the .bin never shows up and the Live Output only shows:

```
Requesting Bluetooth Device...
Connecting to GATT Server...
Getting BlueRetro Service...
app_ver_is_hw2: false app_name_is_hw2: false
Init Cfg DOM...
app_ver_is_hw2: false app_name_is_hw2: false
Init Cfg DOM...
> Bluetooth Device disconnected
```

Thanks!

**Comment by darthcloud** [comment]
What OS and browser are you using?

It's kind weird you got duplicate log message.

**Reply by Kopite11** [reply]
Windows 11 and macOS (Latest versions)
Chrome, Safari and Edge. 

Same result.

I'm thinking about desoldering the ESP32 module and solder it to a DevKit, that way I could try to re-flash it via USB and see if anything changes...

**Comment by darthcloud** [comment]
Unfortunately Retroscaler had the very bad idea to modify the FW. So I got no idea what is running on there.

From the look of it, it look like the adapter crash when you connect to it.

You can just buy a FTDI 3V3 serial usb cable, and connect it to gnd, txd and rxd pins.

**Reply by Kopite11** [reply]
Yeah, it's a bummer that RetroScaler is using a custom firmware for their adapters.

I was just looking the product description on AliExpress, and surely enough, they explicitly advice to not use the official BlueRetro firmware and use theirs instead (This one, in case you want to take a look: https://github.com/RetroScaler/N64-Blueretro)

I'm gonna look into that method you mentioned and see what happens.

Many thanks for your help!

**Comment by darthcloud** [comment]
Can you try again? There was an issue with the OTA page with v1.8 FW that I fixed. I think that was the issue.

**Reply by Kopite11** [reply]
Hello! I'm gonna try again today after work and I'll let you know how it goes :)

**Reply by Kopite11** [reply]
It's working now!

**Reply by FairyKid64** [reply]
I'm having the same issue.  I've been trying to update a RetroScaler N64 adapter that's on firmware 1.8.X from what I remember.  I have been able to connect this adapter before, but not recently.  This is the output I get:

Requesting Bluetooth Device...
Connecting to GATT Server...
Getting BlueRetro Service...
app_ver_is_hw2: false app_name_is_hw2: false
Init Cfg DOM...
\> Bluetooth Device disconnected

**Comment by darthcloud** [comment, marked as answer]
That should now be fixed, try updating again to v1.9.2 using the OTA page.

**Reply by Kopite11** [reply]
Thanks!
